cfme/tests/cloud_infra_common/test_provisioning.py

```diff
@@ -163,36 +163,51 @@ def test_provision_approval(appliance, provider, vm_name, smtp_test, request,
 
     # It will provision two of them
     vm_names = [vm_name + "001", vm_name + "002"]
-    requester = "" if BZ(1628240, forced_streams=['5.10']).blocks else "vm_provision@cfmeqe.com "
+    if BZ(1628240).blocks and provider.one_of(CloudProvider):
+        requester = ""
+    else:
+        requester = "vm_provision@cfmeqe.com "
     collection = appliance.provider_based_collection(provider)
     inst_args = {'catalog': {
         'vm_name': vm_name,
         'num_vms': '2'
     }}
 
     vm = collection.create(vm_name, provider, form_values=inst_args, wait=False)
-    subject = VersionPicker({
-        LOWEST: "your request for a new vms was not autoapproved",
-        "5.10": "your virtual machine request is pending"
-    }).pick()
-    wait_for(
-        lambda:
-        len(filter(
-            lambda mail:
-            subject in normalize_text(mail["subject"]),
-            smtp_test.get_emails())) == 1,
-        num_sec=90, delay=5)
-    subject = VersionPicker({
-        LOWEST: "virtual machine request was not approved",
-        "5.10": "virtual machine request from {}pending approval".format(requester)
-    }).pick()
-    wait_for(
-        lambda:
-        len(filter(
-            lambda mail:
-            subject in normalize_text(mail["subject"]),
-            smtp_test.get_emails())) == 1,
-        num_sec=90, delay=5)
+    try:
+        if provider.one_of(CloudProvider):
+            vm_type = "instance"
+        else:
+            vm_type = "virtual machine"
+
+        subject = VersionPicker({
+            LOWEST: "your request for a new vms was not autoapproved",
+            "5.10": "your {} request is pending".format(vm_type)
+        }).pick()
+        wait_for(
+            lambda:
+            len(filter(
+                lambda mail:
+                normalize_text(subject) in normalize_text(mail["subject"]),
+                smtp_test.get_emails())) == 1,
+            num_sec=90, delay=5)
+        subject = VersionPicker({
+            LOWEST: "virtual machine request was not approved",
+            "5.10": "{} request from {}pending approval".format(vm_type, requester)
+        }).pick()
+
```

Comment:
Optional:
Not required but as we refactor... I would try to iterate these two different `subjects` over `wait_for`.